core/commands/generate_product_repositories_command.rb

```diff
@@ -780,40 +780,55 @@ def create_repository(product)
     info_and_log("Generating repository configuration for #{product}")
     begin
       releases = send("parse_#{product}".to_sym, @config[product])
-    rescue StandardError => error
-      error_and_log("#{product} was not generated. Try again.")
-      error_and_log_error(error)
-      return false
+      if releases.nil?
+        error_and_log("#{product} was not generated. Skiped.")
+        nil
+      else
+        write_product(product, releases)
+        Result.ok(product)
+      end
+    rescue StandardError => e
+      error_and_log("Error message: #{e.message}")
+      error_and_log("#{product} was not generated.")
+      Result.error(product)
     end
-    write_product(product, releases)
-    true
   end
 
   # Print summary information about the created products
   # @param products_with_errors [Array<String>] product names that were not generated
-  def print_summary(products_with_errors)
+  def print_summary(products)
     info_and_log("\n--------\nSUMMARY:\n")
-    @products.sort.each do |product|
-      result = products_with_errors.include?(product) ? '-' : '+'
-      info_and_log("  #{product}: #{result}")
+    products.each do |product|
+      if product.success?
+        info_and_log("  #{product.value}: +")
+      else
+        info_and_log("  #{product.error}: -")
+      end
     end
   end
 
   # Starting point of the application
+  #
+  # The result is equal to nil when additional information from the product is needed
+  # but the information is not in the config
   def execute
     if @env.show_help
       show_help
       return SUCCESS_RESULT
     end
     return ARGUMENT_ERROR_RESULT unless configure_command
+
     remainning_products = @products.dup
-    @attempts.times do
-      remainning_products = remainning_products.reject do |product|
-        create_repository(product)
+    result_products = []
+    result = nil
```

Comment:
If we remove the nil from the method above, may be we could return to the previous implementation?